cpp/katz_centrality_module/algorithm/katz.cpp

```diff
@@ -305,97 +382,100 @@ std::vector<std::pair<std::uint64_t, double>> SetKatz(const mg_graph::GraphView<
     return WrapResults();
   }
 
-  auto deg_max = MaxDegree(graph);
-  double gamma = deg_max / (1. - (alpha * alpha * deg_max));
+  const auto deg_max = MaxDegree(graph);
+  const double gamma = deg_max / (1. - (alpha * alpha * deg_max));
 
   // Initialize the active vector
   std::transform(graph.Nodes().begin(), graph.Nodes().end(),
                  std::inserter(katz_alg::context.active_nodes, katz_alg::context.active_nodes.end()),
-                 [&graph](mg_graph::Node<> vertex) -> std::uint64_t { return graph.GetMemgraphNodeId(vertex.id); });
+                 [&graph](mg_graph::Node<> node) -> std::uint64_t { return graph.GetMemgraphNodeId(node.id); });
 
   return KatzCentralityLoop(katz_alg::context.active_nodes, graph, katz_alg::alpha, katz_alg::k, katz_alg::epsilon,
                             gamma);
 }
 
 std::vector<std::pair<std::uint64_t, double>> UpdateKatz(
-    const mg_graph::GraphView<> &graph, const std::vector<std::uint64_t> &new_vertices,
-    const std::vector<std::pair<std::uint64_t, uint64_t>> &new_edges, const std::vector<std::uint64_t> &new_edge_ids,
-    const std::vector<std::uint64_t> &deleted_vertices,
-    const std::vector<std::pair<std::uint64_t, uint64_t>> &deleted_edges) {
-  // Create context and calculate values if not initialized
-  if (!katz_alg::context.IsInitialized()) {
-    return SetKatz(graph);
+    const mgp::Graph &graph, const std::vector<std::uint64_t> &created_nodes,
+    const std::vector<std::pair<std::uint64_t, uint64_t>> &created_relationships,
+    const std::vector<std::uint64_t> &created_relationship_ids, const std::vector<std::uint64_t> &deleted_nodes,
+    const std::vector<std::pair<std::uint64_t, uint64_t>> &deleted_relationships) {
+  std::uint64_t n_relationships = 0;
+  for (const auto &node : graph.Nodes()) {
+    for (const auto &_ : node.OutRelationships()) {
+      n_relationships++;
+    }
   }
-
-  if (graph.Edges().empty()) {
+  if (n_relationships == 0) {
     katz_alg::context.Init(graph);
     return WrapResults();
   }
 
-  auto deg_max = MaxDegree(graph);
-  double gamma = deg_max / (1. - (alpha * deg_max));
+  const auto deg_max = MaxDegree(graph);
+  const double gamma = deg_max / (1. - (alpha * deg_max));
 
-  // Update context for new vertices for each iteration
-  for (auto v : new_vertices) {
+  // Update context for new nodes for each iteration
+  for (const auto v : created_nodes) {
```

Comment:
do you need reference here

Reply:
We don’t need to reference single `uint64_t`